### PR TITLE
py-graph-tool: Update to version 2.57

### PR DESCRIPTION
**Pull request by essandess**
#### Description

py-graph-tool: Update to version 2.57
    * Add 311, Remove 37
    * Use boost181
    * Bugfix for unary_function C++17 removal

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.4.1 22F770820d arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-graph-tool/Portfile
Comment (ryandesign) [inline python/py-graph-tool/Portfile:86]: In a regular expression `.` means "any character". Did you mean literally a period? If so, escape it with a backslash. Also, did you mean filenames that *contain* `cc` or `hh` or did you mean filenames that *end* with that? If the latter, anchor the regexp to the end of the string with `$`:
```suggestion
                && [regexp "\.(cc|hh)$" [file extension ${f}]] } {
```

File: python/py-graph-tool/Portfile
Comment (ryandesign) [inline python/py-graph-tool/Portfile:88]: Does upstream agree that this is an appropriate fix?

Since you are not interpolating Tcl variables or procedure calls, you can eliminate the need for escaping by using `{}` quoting characters instead of `""`:

```suggestion
                    {s|:[[:space:]]+public[[:space:]]+std::unary_function<.+,.+>||} \
```

File: python/py-graph-tool/Portfile
Comment (ryandesign) [inline python/py-graph-tool/Portfile:136]: Isn't `install` the default value for `destroot.target`?

File: python/py-graph-tool/Portfile
Comment (essandess) [reply python/py-graph-tool/Portfile:86]: Thanks, I used braces: `{\.(cc|hh)$}`. Note that EOL anchor is unnecessary because that's file `file extension` does, but it's good to be explicit for clarity.

File: python/py-graph-tool/Portfile
Comment (essandess) [reply python/py-graph-tool/Portfile:88]: Thanks, done.

File: python/py-graph-tool/Portfile
Comment (essandess) [reply python/py-graph-tool/Portfile:136]: It's overwritten by PG python, so this is necessary here.

File: python/py-graph-tool/Portfile
Comment (essandess) [reply python/py-graph-tool/Portfile:88]: Upstream @count0 is aware of the fix at https://git.skewed.de/count0/graph-tool/-/issues/761. The `unary_function` is unnecessary and has been removed from C++17, so this will certainly work. I'll defer to Tiago how he'd like to address this and have noted in the Portfile that this Portfile-based fix should be removed after the issue is resolved.

**Comment by essandess** [comment]
Note: The CI and buildbots will all fail on this port for `py10-graph-tool` and so forth because of the way the `boost` ports are written: `boost181`, e.g., is always installed with variant `+python11`. There is no way to install the necessary `boost` variants in the Portfile.

cc: @mascguy

**Comment by mascguy** [comment]
> Note: The CI and buildbots will all fail on this port for `py10-graph-tool` and so forth because of the way the `boost` ports are written: `boost181`, e.g., is always installed with variant `+python311`. There is no way to install the necessary `boost` variants in the Portfile.
> 
> cc: @mascguy

I'll try to take a look at a subport approach for 181 soon, as this is definitely a pain...

**Comment by essandess** [comment]
@mascguy May we please merge this PR and get a working `graph-tool` into the field for `arm64` machines?

The `boost` Portfile fixes can be added later.

File: python/py-graph-tool/Portfile
Comment (mascguy) [reply python/py-graph-tool/Portfile:136]: > It's overwritten by PG python, so this is necessary here.

Well, would it be more accurate to say that this is necessary when `python.pep517` is enabled? Which is the default for 3.11 and later.

I'd need to dig into the upstream project a bit, before providing any official recommendations. But if you're working around that by setting `destroot.target`, etc, then perhaps it might be better to simply set `python.pep517` to `no`. (Which might allow you to eliminate at least a few overrides you're setting.)

However, I'll let our more Python-savvy folks chime in with their thoughts and recommendations.

CC: @jmroot
CC: @reneeotten

Comment (mascguy) [reply python/py-graph-tool/Portfile:136]: > I'd need to dig into the upstream project a bit, before providing any official recommendations. But if you're working around that by setting `destroot.target`, etc, then perhaps it might be better to simply set `python.pep517` to `no`. (Which might allow you to eliminate at least a few overrides you're setting.)

That being said, I'm not overly concerned, so no need to hold up merging this.

**Comment by mascguy** [comment]
Presently building the 3.11 subport locally on 10.15 [with trace mode], to see whether all is well. It's taking far longer than expected though.

There are also an inordinate number of warnings, enough so to utterly pollute the build log. I may suppress some of them before merging, as they make it difficult to follow the compilation itself...

**Comment by mascguy** [comment]
> There are also an inordinate number of warnings, enough so to utterly pollute the build log. I may suppress some of them before merging, as they make it difficult to follow the compilation itself.

Which is probably due to the project Makefile setting `-Wall` and `-Wextra`. (With no immediately-apparent way to disable that without patching, based on a quick cursory review of the various AutoTools scripts.)

**Comment by mascguy** [comment]
p.s. You inadvertently included an update to `py-absl`, which you also have an open PR for. So I dropped that commit from this branch.

**Comment by mascguy** [comment]
Building a single subport takes a very long time, at least on my older MacPro. So I've only validated `py311-graph-tool`, but that looks good [with trace mode].

Given that there don't appear to be any dependents on this port, it might be worth dropping subports for 3.9 and prior.

**Review by mascguy — APPROVED** [review]

**Pull request merged**
